abi-versions/v0.2.1/README.md

```diff
@@ -0,0 +1,1807 @@
+
+Called when the Wasm module is first loaded.
+
+
+### `proxy_on_memory_allocate`
+
+* params:
+  - `i32 (size_t) memory_size`
+* returns:
+  - `i32 (uint8_t *) memory_data`
+
+Called to allocate continuous memory buffer of `memory_size` using
+the in-VM memory allocator.
+
+Plugin must return `memory_data` pointing to the start of the allocated
+memory.
+
+Returning `0` indicates failure.
+
+
+### `malloc`
+
+* params:
+  - `i32 (size_t) memory_size`
+* returns:
+  - `i32 (uint8_t *) memory_data`
+
+> **Warning**
+> This is called only in the absence of [`proxy_on_memory_allocate`].
+
+Called to allocate continuous memory buffer of `memory_size` using
+the in-VM memory allocator.
+
+Plugin must return `memory_data` pointing to the start of the allocated
+memory.
+
+Returning `0` indicates failure.
+
+
+## Context lifecycle
+
+### `proxy_on_context_create`
+
+* params:
+  - `i32 (uint32_t) context_id`
+  - `i32 (uint32_t) parent_context_id`
+* returns:
+  - none
+
+Called when the host creates a new context (`context_id`).
+
+When `parent_context_id` is `0` then a new plugin ("root") context
+is created, otherwise a new per-stream context is created.
+
+
+### `proxy_on_done`
+
+* params:
+  - `i32 (uint32_t) context_id`
+* returns:
+  - `i32 (bool) completed`
+
+Called when the host is done processing context (`context_id`).
+
+Plugin must return one of the following values:
+- `true` to allow the host to finalize and delete context.
+- `false` to indicate that the context is still being used,
+  and that plugin is going to call [`proxy_done`] later to
+  allow the host to finalize and delete that context.
+
+
+### `proxy_on_log`
+
+* params:
+  - `i32 (uint32_t) context_id`
+* returns:
+  - none
+
+Called after the host is done processing context, but before
+releasing its state.
+
+This callback can be used for generating final log entries.
+
+
+### `proxy_on_delete`
+
+* params:
+  - `i32 (uint32_t) context_id`
+* returns:
+  - none
+
+Called when the host removes the context (`context_id`).
+
+This is used to signal that the plugin should stop tracking active
+context and remove all associated state.
+
+
+### `proxy_done`
+
+* params:
+  - none
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Indicates to the host that the plugin is done processing active
+context.
+
+This should be used after returning `false` in [`proxy_on_done`].
+
+Returned `status` value is:
+- `OK` on success.
+- `NOT_FOUND` when active context was not pending finalization.
+
+
+### `proxy_set_effective_context`
+
+* params:
+  - `i32 (uint32_t) context_id`
+* returns:
+  - `i32 (`[`proxy_status_t`]`) status`
+
+Changes the effective context to `context_id`.
+
+This can be used to change active context, e.g. during
+[`proxy_on_http_call_response`], [`proxy_on_grpc_receive`]
+and/or [`proxy_on_queue_ready`] callbacks.
+
+Returned `status` value is:
+- `OK` on success.
+- `BAD_ARGUMENT` for unknown `context_id`.
+
+
+## Configuration
+
```

Comment:
add summary comment that these are exported by the guest, and if any are required to be exported.